include/boost/spirit/repository/home/karma/nonterminal/subrule.hpp

```diff
@@ -458,91 +458,35 @@ namespace boost { namespace spirit { namespace repository { namespace karma
                 def_type(compile<spirit::karma::domain>(expr), name_)));
         }
 
-        template <typename Expr>
-        friend typename group_type_helper<Expr, true>::type
-        operator%=(subrule const& sr, Expr const& expr)
-        {
-            typedef group_type_helper<Expr, true> helper;
-            typedef typename helper::def_type def_type;
-            typedef typename helper::type result_type;
-            return result_type(fusion::make_map<id_type>(
-                def_type(compile<spirit::karma::domain>(expr), sr.name_)));
-        }
+#define SUBRULE_MODULUS_ASSIGN_OPERATOR(lhs_ref, rhs_ref)                     \
```

Comment:
Should this have a `BOOST_` prefix? It should be extremely rare to find this macro already declared elsewhere, but ...

Reply:
It had `BOOST_SPIRIT_` but was shortened. Someone even could have and this one, so the actual solution is to add push/pop. I do not want to touch this more, because I have in mind an alternative fix and this overload madness should be removed.